### ISSUE TITLE
Fix issues with Reboot/Restart banner and error popups

### ISSUE THREAD

**Issue by cpinkham** [FalconChristmas/fpp#361]
Issues encountered:

- Error popup on Pi after enabling LED panels and then trying to restart fppd (this restart triggers a reboot to disable the onboard audio, so the restart command fails) http://falconchristmas.com/forum/index.php/topic,8795.msg83211.html#msg83211
- Error popup when expanding SD and then hitting reboot button.  This pops up an error about the restart flag.  http://falconchristmas.com/forum/index.php/topic,8795.msg83211.html#msg83211

**Comment by jaredb7** [comment]
I've noticed the rebootFlag issue is most likely happening due to possibly session locking or some other delay seeing as all the commands are fired off at once. 

You occasionally have the Clear Restart Flag go through, Clear Reboot Flag request is still running, and rebootPi request reboots the Pi, Apache shuts down (some 500 error) and setSettings (from Clear Reboot) the will go through it's fail callback & display the dialog.

This could be tackled from the front end or backend, I've added a setTimeout to delay the reboot 1 second after the clear restart & reboot flags which appears to work. Any delay will work as long as we can add some kind of buffer between the clearing flags and the actual reboot.
Or we could change the command in fppxml to delay the reboot 1 second somehow.


With the FPPD restarting the FPP system, I implemented a retry system which might be overkill, but every 2 seconds it polls the isFPPDrunning xml endpoint to basically find out if the FPP has restarted (I'm doing this in the fail call-back, it's likely we're here due to request timing out or terminating because of the reboot). 

Once it's up, clears the restartFPPD flag (which would of already been reset during the reboot anyway).
Then runs the code from the success callback which clears the restart banner without having to refresh the page. 

I also pop a notification that the FPP is rebooting which hangs around for 10 seconds just to try let the user know what's happening.

![image](https://user-images.githubusercontent.com/799998/38367040-ef1fdd1c-3924-11e8-8ad1-c645e5ad644e.png)
